### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by lurch** [davidferguson/pibakery-blocks#8]
Some other nice "first boot" blocks might be some of the stuff that you typically use `raspi-config` to do just once, e.g. setting timezone, enabling camera, setting localisation, setting memory split, etc. etc.
You could look at https://github.com/RPi-Distro/rc_gui/blob/master/src/rc_gui.c to see how to 'drive' raspi-config using just command-line options.

**Comment by davidferguson** [comment]
Thanks - I'll look into it.

**Comment by dkirkby** [comment]
I agree this would be useful.  It looks like you already have one block (Set hostname) that calls raspi-config and another (Set Boot Option) that is mostly copied from raspi-config.  However, using one block per raspi-config option probably does not scale well, especially with all blocks currently in the same top-level directory.

**Comment by lurch** [comment]
> However, using one block per raspi-config option

From what little I know of pibakery, that's the way it _should_ work (although probably not necessary to cover **every** raspi-config option?)

> especially with all blocks currently in the same top-level directory.

I wasn't aware of that. Maybe have different subdirectories for different categories?

**Comment by davidferguson** [comment]
>From what little I know of pibakery, that's the way it should work (although probably not necessary to cover every raspi-config option?)

That's right, and I agree that not every option from raspi-config is needed (expand filesystem is redundant anyway, I believe the camera is auto-enabled now, etc).

The way PiBakery loads blocks at the moment requires them to all be in one directory. I can't see this being a problem in the immediate future, as I've worked with directories that contain many, many, many more directories than that (8 million at the last count).

**Comment by dkirkby** [comment]
With 5-10 raspi-config options, I agree that one per block is best for the new user but if you implemented all possible options (which someone will eventually ask for), the sheer number could be overwhelming unless there is some way to organize them hierarchically, and separate out the less common options.

**Comment by davidferguson** [comment]
If you're talking about how the blocks are viewed in Blockly, there is a way for me to set the order that blocks are displayed in each category in Blockly, and of course I can also add (and remove) categories as needed.

If you're talking about how they're saved in the `pibakery-blocks` directory, there's not much I can do about that now.

**Comment by lurch** [comment]
> there's not much I can do about that now.

As an outside observer, it's great to see you putting so much emphasis on maintaining backwards-compatibility :+1:

**Comment by dkirkby** [comment]
As another outside observer, the overall idea and execution of this project is great - thanks!!

**Comment by davidferguson** [comment]
I've added in blocks for Set SPI and Set I2C - any other suggestions from raspi-config?

**Comment by dkirkby** [comment]
I vote for setting the locale, timezone and keyboard map although the raspi-config menus are not very user friendly for this.  It would be great to provide two versions, e.g. basiclocalize and expertlocalize, where basiclocalize has a single menu parameter of common choices (UK, USA, ...) and takes care of all three settings, while expertlocalize has three text parameters and requires that you know what you are doing.

**Comment by cognitivegears** [comment]
FYI I just added a pull request for memsplit as I had need of this on a project I'm working on.  I might take a try at the other ones mentioned above too if nobody else grabs them first.

**Comment by MaitreDede** [comment]
Hi :)

My suggestions :

* Internationalization
* GPIO Server (advanced options)
* Audio (force output to jack, hdmi or auto)

**Comment by rmayol** [comment]
First of all: Thank you and congratulations for PiBakery!!
One question: I enable VPN server, but I don't know wich username and password it has configured. I configured RealVPN viewer on Mac and when I try to connect, it refuse my connection without more information.

**Comment by lurch** [comment]
Did you mean "VNC" instead of "VPN"? If you did, have a look at https://www.raspberrypi.org/documentation/remote-access/vnc/README.md

**Comment by eddwatts** [comment]
i have an audio block (submitted for a pull request along with a few others), force to jack or hdmi but not the auto setting:
https://github.com/eddwatts/pibakery-blocks/tree/master/setaudio

**Comment by nsayer** [comment]
In addition to setting the locale, a good idea would be an option to set the WiFi regulatory domain.

**Comment by eddwatts** [comment]
WiFi regulatory domain:
have done this myself a few months ago - just not been added to the main section
my updated block:
https://github.com/eddwatts/pibakery-blocks/tree/master/wifisetup

**Comment by dannystaple** [comment]
Hmmm - did a fresh PiBakery write today - sadly looks like Camera is not enabled by default:

`picamera.exc.PiCameraError: Camera is not enabled. Try running 'sudo raspi-config' and ensure that the camera has been enabled.`
